sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -358,4 +358,30 @@ class DataSourceWithHiveMetastoreCatalogSuite
         Seq(table("src").count().toString))
     }
   }
+
+  test("SPARK-29899: Recursively load data in table via TBLPROPERTIES") {
+    withTempPath(dir => {
+      val baseDir = s"${dir.getCanonicalFile.toURI.toString}/path1"
+      val innerDir = s"$baseDir/path2/path3"
+      spark.range(3).selectExpr("id").write.parquet(innerDir)
+      withTable("test1", "test2") {
+        withSQLConf(HiveUtils.CONVERT_METASTORE_PARQUET.key -> "true") {
+          spark.sql(
+            s"""
+               |CREATE TABLE test1 (id bigint)
+               |STORED AS PARQUET LOCATION '$baseDir'
+               |TBLPROPERTIES (
+               | 'recursiveFileLookup'='true')
```

Comment:
Sorry to ask tangential questions, but I'm curious: Will the Metastore track this property somehow? i.e. If I create a table with `'recursiveFileLookup'='true'` using Spark, can I query it from Presto and see the same data, provided that both are pointed at the same Metastore? Will the Metastore just track the table property, or will it also track the list of data paths that were detected when the table was created or refreshed?

Reply:
Thanks to point me this. Maybe 'spark.recursiveFileLookup' is much more meaningful for user.